test/new-e2e/npm/ec2_1host_test.go

```diff
@@ -0,0 +1,172 @@
+network_config:
+  enabled: true
+`
+
+type ec2VMSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+	DevMode bool
+}
+
+// TestEC2VMSuite will validate running the agent on a single EC2 VM
+func TestEC2VMSuite(t *testing.T) {
+	s := &ec2VMSuite{}
+	e2eParams := []params.Option{}
+	// debug helper
+	if _, devmode := os.LookupEnv("TESTS_E2E_DEVMODE"); devmode {
+		e2eParams = []params.Option{params.WithDevMode(), params.WithSkipDeleteOnFailure()}
+		s.DevMode = true
+	}
+
+	// Source of our kitchen CI images test/kitchen/platforms.json
+	// Other VM image can be used, our kitchen CI images test/kitchen/platforms.json
+	// ec2params.WithImageName("ami-a4dc46db", os.AMD64Arch, ec2os.AmazonLinuxOS) // ubuntu-16-04-4.4
+	e2e.Run(t, s, e2e.FakeIntakeStackDef(nil, agentparams.WithSystemProbeConfig(NPMsystemProbeConfig)), e2eParams...)
+}
+
+// TestFakeIntakeNPM Validate the agent can communicate with the (fake) backend and send connections every 30 seconds
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+func (v *ec2VMSuite) TestFakeIntakeNPM() {
+	t := v.T()
+
+	// default is to reset the current state of the fakeintake aggregators
+	if !v.DevMode {
+		v.Env().Fakeintake.FlushServerAndResetAggregators()
+	}
+
+	targetHostnameNetID := ""
+	// looking for 1 host to send CollectorConnections payload to the fakeintake
+	err := backoff.Retry(func() error {
+		// generate a connection
+		v.Env().VM.Execute("curl http://www.datadoghq.com")
+
+		hostnameNetID, err := v.Env().Fakeintake.GetConnectionsNames()
+		if err != nil {
+			return err
+		}
+		if len(hostnameNetID) == 0 {
+			return errors.New("no connections yet")
+		}
+		targetHostnameNetID = hostnameNetID[0]
+
+		t.Logf("hostname+networkID %v seen connections", hostnameNetID)
+		return nil
+	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(time.Second), 60))
+	require.NoError(t, err)
+
+	// looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+	err = backoff.Retry(func() error {
+		cnx, err := v.Env().Fakeintake.GetConnections()
+		require.NoError(t, err)
+
+		payloadsTimestamp := cnx.GetCollectedTimeByName(targetHostnameNetID)
+		if len(payloadsTimestamp) < 3 {
+			return errors.New("not enough payloads")
+		}
+
+		dt := float64(payloadsTimestamp[2].Sub(payloadsTimestamp[1])) / float64(time.Second)
+		t.Logf("hostname+networkID %v diff time %f seconds", targetHostnameNetID, dt)
+		require.LessOrEqual(t, math.Abs(dt-30), 0.5, "delta between collection is higher that 500ms")
+		return nil
+	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(time.Second), 90))
+	require.NoError(t, err)
+}
+
+// TestFakeIntakeNPM_TCP_UDP_DNS validate we received tcp, udp, and DNS connections
+// with some basic checks, like IPs/Ports present, DNS query has been captured, ...
+func (v *ec2VMSuite) TestFakeIntakeNPM_TCP_UDP_DNS() {
+	t := v.T()
+
+	err := backoff.Retry(func() error {
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
Same as above, I would rather act and apply retries only on the fakeintake, or have different retry strategies based on act vs assert.